### DISCUSSION TITLE
Welcome to competition-management-system Discussions!

### DISCUSSION THREAD

**Discussion by priyansh32** [NITK-KODE/cms-api#2] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by priyansh32** [comment]
For the maintainers, here are some tips 💡 for getting started with Discussions. We'll leave these in Markdown comments for now, but feel free to take out the comments for all maintainers to see.

  📢 **Announce to your community** that Discussions is available! Go ahead and send that tweet, post, or link it from the website to drive traffic here.

  🔗 If you use issue templates, **link any relevant issue templates** such as questions and community conversations to Discussions. Declutter your issues by driving community content to where they belong in Discussions. If you need help, here's a [link to the documentation](https://docs.github.com/en/github/building-a-strong-community/configuring-issue-templates-for-your-repository#configuring-the-template-chooser).

  ➡️ You can **convert issues to discussions** either individually or bulk by labels. Looking at you, issues labeled “question” or “discussion”.

**Comment by priyansh32** [comment]
Getting this response when trying to `GET /events`
```json
{
    "success": false,
    "message": "func is not a function"
}
```
![image](https://user-images.githubusercontent.com/75668169/133787787-098754ea-10a9-488b-8c6f-4178c1b246ab.png)

**Reply by Utkar5hM** [reply]
Oh there's a mistake in the code in /routes/independent/events.js.

it should be Index instead of index.

**Reply by priyansh32** [reply]
I resolved that, still getting this

**Reply by Utkar5hM** [reply]
![image](https://user-images.githubusercontent.com/14823803/133789067-6efd0a68-9a5c-4e5c-bd78-2ae879cea6d4.png)
Seems to work for me.

**Reply by priyansh32** [reply]
there is some problem with this function, we need to pass func in param in catchAsync
```js
module.exports = (func) => (req, res, next) => {
  func(req, res, next).catch(next);
};
```

**Reply by BenzeneAlcohol** [reply]
One important thing I found was we are not using try-catch anywhere. Whenever there is an async function, it is good practice to use try-catch, in my opinion.

**Reply by priyansh32** [reply]
catchAsync is used only for that ig

**Reply by BenzeneAlcohol** [reply]
Oh, my bad, did not notice it in the utilities!

**Reply by Utkar5hM** [reply]
Isn't it what we are doing right now?

**Reply by BenzeneAlcohol** [reply]
Yes, you are right. I was wondering how it worked without try-catch, now saw it in utils.

**Reply by priyansh32** [reply]
@Utkar5hM  user needs to be logged in to fetch events?
also I am not able to register user
![image](https://user-images.githubusercontent.com/75668169/133789767-6e7762d6-7105-4fa2-843f-68783d1ec301.png)

**Reply by Utkar5hM** [reply]
It does accept in the form of x-www-form-urlencoded . Currently in your case its not taking your input. @priyansh32

**Reply by priyansh32** [reply]
oh, when making requests from frontend we will probably use json
should I add express json parser?

**Reply by priyansh32** [reply]
It's working now! thanks

**Reply by Utkar5hM** [reply]
> oh, when making requests from frontend we will probably use json
> should I add express json parser?

That's fine

**Comment by priyansh32** [comment]
Should we create a postman workspace?

**Reply by Utkar5hM** [reply]
@roshgupta,  The postman team is currently full as it supports only 3 people at max  in the free plan.
